internal/kms/kmip.go

```diff
@@ -0,0 +1,527 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kms
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/ceph/ceph-csi/internal/util/k8s"
+
+	kmip "github.com/gemalto/kmip-go"
+	"github.com/gemalto/kmip-go/kmip14"
+	"github.com/gemalto/kmip-go/ttlv"
+	"github.com/google/uuid"
```

Comment:
we use github.com/pborman/uuid already, maybe create a new issue to move to github.com/google/uuid instead? The google one is used by IBM Keyprotect and Kubernetes packages already, pborman only in `internal/journal/voljournal.go`

Reply:
https://github.com/ceph/ceph-csi/issues/3315

Reply:
ok, thanks!